KubeArmor/core/kubeUpdate.go

```diff
@@ -509,6 +509,66 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 	}
 }
 
+func getTopLevelOwner(obj metav1.ObjectMeta, namespace string, objkind string) (string, string, string, error) {
+	ownerRef := kl.GetControllingPodOwner(obj.OwnerReferences)
+	if ownerRef == nil {
+		return obj.Name, objkind, namespace, nil
+	}
+
+	switch ownerRef.Kind {
+	case "Pod":
+		pod, err := K8s.K8sClient.CoreV1().Pods(namespace).Get(context.Background(), ownerRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return "", "", "", err
+		}
+		if len(pod.OwnerReferences) > 0 {
+			return getTopLevelOwner(pod.ObjectMeta, namespace, "Pod")
+		}
```

Comment:
This case will not occur for our use case.
Ref - https://accuknox.slack.com/archives/C0229TD83RA/p1688386506276139

For a static Pod `ownerRef` would be nil. 
For ex check how it is implemented here https://github.com/kubearmor/KubeArmor/blob/main/KubeArmor/core/kubeUpdate.go#L584

Reply:
this is not for static pods

Reply:
can you give an example when Pod could be an owner of an kubernetes object related to our use case here? 
I'm not sure whether this could be useful here.

Reply:
:thinking: umm i cant find particular example for this case 
but i thought of this case as kiem tool would run as a job and will be created by another pod so the owner would be that pod.

Reply:
now that i think about it, this analogy doesnt make sense :sweat_smile: .
will this create any error ?
if not should we let it be there just in case ?

Reply:
Ohk, this won't create any error, I was just ensuring that we should not have anything unnecessary, anyways we can keep  it.